libs/shared/src/main/java/io/crate/data/Killable.java

```diff
@@ -25,6 +25,11 @@
 
 public interface Killable {
 
+    enum Type {
+        HARD,
+        SOFT
+    }
+
```

Comment:
I this these could use some docs to explain the difference in behavior

Reply:
I'm also not entirely sure about the naming here. I feel like they don't capture the semantics in a way that explains their difference. What's a soft kill and a hard kill?

If I understand the intention right, it is mostly about preserving parts of the results instead of failing with a "job killed". Maybe we could capture this somehow?

Reply:
Thanks I agree, the intention was to make the target iterator or task to stall or do nothing, instead of throwing from it. I renamed it to `SUPPRESS_EXCEPTION`